### DISCUSSION TITLE
Is it possible to get layer(primitives/roots) name like materials name function?

### DISCUSSION THREAD

**Discussion by mustafa-soylemez** [google/model-viewer#4662] [Q&A] [answered]
Is there a defined function or a way to get layer name from clicked position on model like materialFromPoint function?

http://prnt.sc/uPvnjhWA_Mvz

**Comment by elalish** [comment]
Not sure what you mean by "layer", but we don't expose nodes/meshes (or their names) in our API. You can get the material and its name though.

**Reply by mustafa-soylemez** [reply]
What I mean with layer is names here: http://prnt.sc/uPvnjhWA_Mvz

I'm trying to get these names from clicking on meshes like how materialFromPoint function works.

**Reply by elalish** [reply, marked as answer]
No, those are node/mesh names and we don't expose such a low level API here. We recommend making separate materials for anything you want independently selectable, and then use `materialFromPoint`.
